plugins/source/aws/resources/services/kms/grants_fetch.go

```diff
@@ -0,0 +1,34 @@
+package kms
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/kms"
+	"github.com/aws/aws-sdk-go-v2/service/kms/types"
+	"github.com/cloudquery/cloudquery/plugins/source/aws/client"
+	"github.com/cloudquery/plugin-sdk/schema"
+)
+
+func fetchKmsGrants(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- interface{}) error {
+	k := parent.Item.(types.KeyMetadata)
+	config := kms.ListGrantsInput{KeyId: k.KeyId}
+
+	c := meta.(*client.Client)
+	svc := c.Services().KMS
+	p := kms.NewListGrantsPaginator(svc, &config)
+	for p.HasMorePages() {
+		response, err := p.NextPage(ctx)
+		if err != nil {
+			return err
+		}
+		res <- response.Grants
+	}
+	return nil
+}
+
+func resolveGrantARN(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource, c schema.Column) error {
+	cl := meta.(*client.Client)
+	g := resource.Item.(types.GrantListEntry)
+	return resource.Set(c.Name, cl.ARN("kms", "grant", aws.ToString(g.KeyId), aws.ToString(g.GrantId)))
```

Comment:
Another made up ARN.

Reply:
Probably worth creating a grant in a test account just to check that the ARN matches

Reply:
The grants don't have arns so it's made up just like few others. Nothing to check in a test account.

Reply:
If it doesn't match anything real on the AWS side, then I think we shouldn't be adding the ARN column? I wasn't aware we're doing this in other places as well

Reply:
@hermanschaaf check "Resource Issues" in [#3067 ](https://github.com/cloudquery/cloudquery/issues/3067)

Reply:
Okay but most of those do exist, they're just wrong. And the ones that don't exist should be removed, let's not add more non-existent ones 🤷  If the issue is that we need a PK for upserting, we can use the separate columns as a composite PK, which I guess would be account_id, region_id, key_id, grant_id in this case.

Reply:
Look in the "Resource Issues" not the whole issue description. Instance statuses etc.

Reply:
removed in [`b68dd67` (#3441)](https://github.com/cloudquery/cloudquery/pull/3441/commits/b68dd671c66e3fd147891b4ded8a9d5178351e8c)

Reply:
@disq I read the issue, and saw 3 instances of made-up ARNs discovered there. I don't think that's a good enough reason to add made-up ARNs as a rule now, so I'm not sure what you're saying. Let's rather remove any made-up ARNs that we become aware of. Unless there is some other reason to add it that I'm missing, that is. But even even if we do need a unique identifier for some reason, we shouldn't call it `arn` if it doesn't match something on the Amazon side.

Reply:
Also #3466

Reply:
I don't see an ARN for KMS Grants either.  The GrantID looks to be a unique identifier as specified by AWS in their documentation - so we can use that as the unique identifier: https://docs.aws.amazon.com/kms/latest/APIReference/API_GrantListEntry.html.  

Will test later to check for an ARN.